### DISCUSSION TITLE
Redraw a StatusItem in the StatusBar

### DISCUSSION THREAD

**Discussion by ghost** [tui-cs/Terminal.Gui#2329] [Q&A] [answered]
Apologies for such a mundane question, but is there an efficient way to redraw an individual `StatusItem` in the `StatusBar` instead of forcing an entire window refresh?

**Comment by tznind** [comment, marked as answer]
I don't think you can refresh an individual item but you can use `StatusBar.SetNeedsDisplay()` to redraw the whole bar (but not rest of the UI).

If you are doing this from a Thread though I would make sure to call wrapped with `Application.MainLoop.Invoke(()=>myStatusBar.SetNeedsDisplay());`.
